errors/material/task.md

```diff
@@ -0,0 +1,73 @@
+
+Zu diesem Zweck wurden uns freundlicherweise bereits einige `txt`-Dateien zur Verfügung gestellt, die die auf verschiedenen Seiten enthaltenen Wörter beinhalten. Zum Beispiel kann in der Datei `words-lectures-4-memory-notes4.txt` pro Zeile ein Wort aus [Vorlesungsnotizen Woche 4](https://inf.zone/lectures/4-memory/notes4/) gefunden werden. Es kann davon ausgegangen werden, dass jedes Wort maximal _255 Zeichen lang_ ist und nur aus _kleinen ASCII Buchstaben_ besteht. Umlaute wie _ä_, _ü_ oder _ö_ sind bereits durch _ae_, _ue_ und _oe_ ersetzt worden. Ebenfalls müssen eingegebene Wörter, welche diesen Voraussetzungen nicht entsprechen auch _nicht gesondert behandelt werden_. Gibt ein Benutzer also z.B. `Hallü` in das Programm ein, kann es abstürzen, weiterlaufen oder eine Fehlermeldung ausgeben, was hierbei getan wird, liegt an dir.
+
+Um diese einfachere Suche zu ermöglichen, hat ein ungenannter Tutor bereits versucht, ein solches Programm in C zu entwickeln. Dieses ist in der Datei `search-zone.c` zu finden.
+**Es enthält jedoch VIELE Fehler und ist daher absolut kein vorbildliches oder funktionierendes Beispiel!**
+
+::: danger
+Der Code dieses Programms ist auch nicht sonderlich sicher und daher anfällig für diverse Angriffe.
+Für die Zwecke von Inf-Einf reicht das hier aber völlig aus.
+:::
+
+## Teilaufgabe 1
+
+Verschaffe dir einen Überblick über das Programm des Tutors, beachte hierbei allerdings dass es einige syntaktische[^1] aber auch logische[^2] Fehler enthält.
+Beantworte hierzu insbesondere die folgenden Fragen:
+
+1. Welche Datenstruktur wird verwendet um die Wörter zu speichern und zu suchen? _Warum genau diese?_
+2. Was ist die Aufgabe der einzelnen Funktionen des Programms? _Schreibe Kommentare zu diesen!_
+3. Wofür wird die Funktion `free_node` benötigt?
+
+[^1]: **Syntaktische Fehler** sind solche, die durch den Compiler erkannt werden können (z.B. ein fehlendes `;`). Wir können sie daran erkennen, dass sich das Programm nicht kompilieren lässt und `make solution-zone` einige Fehlermeldungen ausgibt.
+[^2]: **Logische Fehler** sind Fehler des Programmablaufes (z.B dass in einer Funktion `int add(int a, int b)`, statt zu addieren subtrahiert wird.). Wir können sie daran erkennen, dass sich unser Programm komisch/unerwartet verhält, wenn wir es ausführen.
+
+## Teilaufgabe 2
+
+Nachdem wir uns in [Teilaufgabe 1](#teilaufgabe-1) einen groben Überblick über das Programm beschafft haben, ist es nun an der Zeit die Fehler zu beheben.
+Vergiss nicht, das Programm enthält sowohl syntaktische als auch logische Fehler enthält.
+
+## Teilaufgabe 3
+
+Da das Programm dank [Teilaufgabe 2](#teilaufgabe-2) nun endlich funktioniert, könnten wir den Code noch verbessern.
+Insbesondere, da unter anderem einige [Magic Numbers](https://inf.zone/lectures/1-c/short-1-6-magic-numbers/) verwendet werden.
```

Comment:
I do not get the point here just from reading the description. Do you mean that we should add preprocessor directives?
Then we should state it in an more obvious way.